app/k9mail/src/main/java/com/fsck/k9/widget/unread/UnreadWidgetConfigurationFragment.kt

```diff
@@ -32,6 +32,27 @@ class UnreadWidgetConfigurationFragment : PreferenceFragmentCompat() {
     private var selectedAccountUuid: String? = null
     private var selectedFolderId: Long? = null
 
+    override fun onSaveInstanceState(outState: Bundle) {
+            // Use local variable to avoid mutable property issues.
+            // putString supports null storage, but we will type check for consistency
+            var localSelectedAccountUuid = selectedAccountUuid
+            var localSelectedFolderId = selectedFolderId
+            var localSelectedFolderSummary = unreadFolder.summary.toString()
```

Comment:
Rather than saving the summary value of the `Preference` please write the folder display name to a property in `handleChooseFolder()` and use that property when saving state.